### ISSUE TITLE
Loss is negative？

### ISSUE THREAD

**Issue by changwh** [ifzhang/FairMOT#38]
Hello, I ran the `ft_mot20_dla34.sh`, and got following log:
```
2020-04-15-15-52: epoch: 1 |loss 8.929329 | hm_loss 0.650758 | wh_loss 0.932686 | off_loss 0.215589 | id_loss 5.410082 | time 38.966667 | 
2020-04-15-16-31: epoch: 2 |loss 4.626980 | hm_loss 0.527663 | wh_loss 0.815603 | off_loss 0.208662 | id_loss 2.909954 | time 38.533333 | 
2020-04-15-17-09: epoch: 3 |loss 2.803859 | hm_loss 0.488239 | wh_loss 0.790705 | off_loss 0.205520 | id_loss 1.759640 | time 38.516667 | 
2020-04-15-17-48: epoch: 4 |loss 1.848163 | hm_loss 0.462011 | wh_loss 0.775284 | off_loss 0.204124 | id_loss 1.153706 | time 38.516667 | 
2020-04-15-18-27: epoch: 5 |loss 1.303856 | hm_loss 0.443506 | wh_loss 0.763248 | off_loss 0.202581 | id_loss 0.820189 | time 38.566667 | 
2020-04-15-19-05: epoch: 6 |loss 0.945925 | hm_loss 0.425916 | wh_loss 0.745333 | off_loss 0.201435 | id_loss 0.612418 | time 38.683333 | 
2020-04-15-19-44: epoch: 7 |loss 0.714345 | hm_loss 0.414678 | wh_loss 0.742630 | off_loss 0.200537 | id_loss 0.482441 | time 38.816667 | 
2020-04-15-20-23: epoch: 8 |loss 0.536990 | hm_loss 0.403566 | wh_loss 0.732034 | off_loss 0.199769 | id_loss 0.390953 | time 38.716667 | 
2020-04-15-21-02: epoch: 9 |loss 0.394856 | hm_loss 0.394498 | wh_loss 0.727754 | off_loss 0.199561 | id_loss 0.323789 | time 38.700000 | 
2020-04-15-21-40: epoch: 10 |loss 0.266375 | hm_loss 0.385873 | wh_loss 0.718381 | off_loss 0.198318 | id_loss 0.271420 | time 38.750000 | 
2020-04-15-22-19: epoch: 11 |loss 0.160874 | hm_loss 0.380030 | wh_loss 0.719103 | off_loss 0.197896 | id_loss 0.233149 | time 38.733333 | 
2020-04-15-22-58: epoch: 12 |loss 0.060106 | hm_loss 0.373991 | wh_loss 0.711818 | off_loss 0.197058 | id_loss 0.199241 | time 38.733333 | 
2020-04-15-23-37: epoch: 13 |loss -0.018475 | hm_loss 0.369472 | wh_loss 0.711355 | off_loss 0.197177 | id_loss 0.176589 | time 38.733333 | 
2020-04-16-00-15: epoch: 14 |loss -0.094603 | hm_loss 0.365172 | wh_loss 0.702102 | off_loss 0.196040 | id_loss 0.155684 | time 38.733333 | 
2020-04-16-00-54: epoch: 15 |loss -0.153388 | hm_loss 0.363450 | wh_loss 0.704547 | off_loss 0.196285 | id_loss 0.140758 | time 38.766667 | 
2020-04-16-01-33: epoch: 16 |loss -0.230269 | hm_loss 0.356688 | wh_loss 0.691093 | off_loss 0.195501 | id_loss 0.116214 | time 38.766667 | 
2020-04-16-02-12: epoch: 17 |loss -0.236539 | hm_loss 0.356235 | wh_loss 0.694913 | off_loss 0.195542 | id_loss 0.114774 | time 38.800000 | 
2020-04-16-02-51: epoch: 18 |loss -0.246128 | hm_loss 0.355469 | wh_loss 0.692207 | off_loss 0.195600 | id_loss 0.112248 | time 38.783333 | 
2020-04-16-03-30: epoch: 19 |loss -0.255631 | hm_loss 0.354308 | wh_loss 0.689188 | off_loss 0.195229 | id_loss 0.110086 | time 38.833333 | 
2020-04-16-04-08: epoch: 20 |loss -0.264329 | hm_loss 0.353399 | wh_loss 0.686179 | off_loss 0.194963 | id_loss 0.108142 | time 38.850000 | 
```
As we can see, the loss is negative after the 13th epoch. Is it normal? And is the 12th model is the best according the log?

**Comment by ifzhang** [comment]
It is normal because we use the uncertainty loss function. The total loss can be negative when other losses are small. I think the 20th model may be the best and you can try it on the training set of MOT20.

**Comment by changwh** [comment]
> 
> 
> It is normal because we use the uncertainty loss function. The total loss can be negative when other losses are small. I think the 20th model may be the best and you can try it on the training set of MOT20.

Thanks for your quick reply. I will have a try.

**Comment by shupinghu** [comment]
> > It is normal because we use the uncertainty loss function. The total loss can be negative when other losses are small. I think the 20th model may be the best and you can try it on the training set of MOT20.
> 
> Thanks for your quick reply. I will have a try.

Hello, I wonder could you teach me how you get such a low loss? I'm trying to train the dla34 model, but the reid loss can only reach to 5

**Comment by changwh** [comment]
> 
> 
> > > It is normal because we use the uncertainty loss function. The total loss can be negative when other losses are small. I think the 20th model may be the best and you can try it on the training set of MOT20.
> > 
> > 
> > Thanks for your quick reply. I will have a try.
> 
> Hello, I wonder could you teach me how you get such a low loss? I'm trying to train the dla34 model, but the reid loss can only reach to 5

Just run the `ft_mot20_dla34.sh`? It a little far from now, I am not comfirmed whether I modified some things, maybe removed the `MOT20-05` data from training set or not. But I got negative loss when the training finished(20 epoches).
